barretenberg/cpp/src/barretenberg/vm/avm_trace/AvmMini_trace.cpp

```diff
@@ -544,14 +576,17 @@ void AvmMiniTraceBuilder::internal_return()
 std::vector<Row> AvmMiniTraceBuilder::finalize()
 {
     auto mem_trace = mem_trace_builder.finalize();
```

Comment:
this is a comment about line 503: `internal_call`

are we supporting a JMP instruction? If so this opens up a potential issue regarding call stack manipulation.

e.g. function `A` calls `B` which calls `C`.

If `C` unconditionally jumps back into `B`, then when `B` returns, the return destination pulled off of the call stack will still contain the return details for call `B->C`, not `A->B` 

I don't know if this is something to address as it requires the program writer to do some very stupid things to trigger this case, but it might be worth discussing + making an issue about?

Reply:
@Maddiaa0 This is about some of your code. Would you like to handle this?